src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java

```diff
@@ -341,4 +342,45 @@ public static void validateArraySchema(Schema arraySchema, String name) {
                                                        name, componentSchema));
     }
   }
+
+  public static void validateColumnForPartition(String columnName, Schema outputSchema) {
+    if (columnName == null) {
+      return;
+    }
+    if (outputSchema == null) {
```

Comment:
If the schema contains a macro, it will be null during the configurePipeline() call. In that scenario, this should not throw an exception. This is one reason this method fits better in the config class rather than here, as there you can check with containsMacro("schema")

Reply:
Done.